aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -1804,7 +1804,7 @@ impl AptosVM {
         type_args: Vec<TypeTag>,
         arguments: Vec<Vec<u8>>,
         gas_budget: u64,
-    ) -> Result<Vec<Vec<u8>>> {
+    ) -> Result<(Vec<Vec<u8>>, u64)> {
```

Comment:
Personally, it would be preferred to be a struct, so the outputs are known names.  That way it won't require someone to read the code at the end of the function to determine it's the gas used.

Reply:
+1, we can have something similar to `TransactionOutput` or `VMOutput` structs